daffodil-runtime2/src/main/resources/org/apache/daffodil/runtime2/c/libruntime/infoset.h

```diff
@@ -26,22 +26,26 @@
 
 // Prototypes needed for compilation
 
-typedef struct ElementRuntimeData ERD;
-typedef struct InfosetBase        InfosetBase;
-typedef struct PState             PState;
-typedef struct UState             UState;
-typedef struct VisitEventHandler  VisitEventHandler;
-
-typedef void (*ERDInitSelf)(InfosetBase *infoNode);
-typedef void (*ERDParseSelf)(InfosetBase *infoNode, PState *pstate);
-typedef void (*ERDUnparseSelf)(const InfosetBase *infoNode, UState *ustate);
-typedef const Error *(*InitChoiceRD)(const InfosetBase *infoNode, const InfosetBase *rootElement);
-
-typedef const Error *(*VisitStartDocument)(const VisitEventHandler *handler);
-typedef const Error *(*VisitEndDocument)(const VisitEventHandler *handler);
-typedef const Error *(*VisitStartComplex)(const VisitEventHandler *handler, const InfosetBase *base);
-typedef const Error *(*VisitEndComplex)(const VisitEventHandler *handler, const InfosetBase *base);
-typedef const Error *(*VisitNumberElem)(const VisitEventHandler *handler, const ERD *erd, const void *number);
+struct ERD;
```

Comment:
Surprising to me that this typedef isn't portable. Seriously you can't suppress the need to use the struct keyword everywhere you use these types? Ugh. 

Can we upgrade our usage to a newer still-standard revision of the C language that has this?